src/modules/75-cd/components/PipelineSteps/TerraformCloudRollbackStep/TerraformCloudRollback.tsx

```diff
@@ -0,0 +1,387 @@
+                  name="spec.discardPendingRuns"
+                  label={getString('pipeline.terraformStep.discardPendingRuns')}
+                  disabled={readonly}
+                  multiTypeTextbox={{ expressions, allowableTypes }}
+                  className={css.addMarginTop}
+                />
+                {getMultiTypeFromValue(values.spec?.discardPendingRuns) === MultiTypeInputType.RUNTIME && (
+                  <ConfigureOptions
+                    value={(values.spec?.discardPendingRuns || '') as string}
+                    type="String"
+                    variableName="spec.discardPendingRuns"
+                    showRequiredField={false}
+                    showDefaultField={false}
+                    showAdvanced={true}
+                    onChange={value => setFieldValue('spec.discardPendingRuns', value)}
+                    style={{ alignSelf: 'center', marginTop: 11 }}
+                    className={css.addMarginTop}
+                    isReadonly={readonly}
+                  />
+                )}
+              </div>
+
+              <div className={cx(stepCss.formGroup, stepCss.md)}>
+                <FormInput.MultiTextInput
+                  name="spec.provisionerIdentifier"
+                  placeholder={getString('pipeline.terraformStep.provisionerIdentifier')}
+                  label={getString('pipelineSteps.provisionerIdentifier')}
+                  multiTextInputProps={{ expressions, allowableTypes }}
+                  disabled={readonly}
+                  className={css.addMarginTop}
+                />
+                {getMultiTypeFromValue(values.spec.provisionerIdentifier) === MultiTypeInputType.RUNTIME && (
+                  <ConfigureOptions
+                    value={values.spec.provisionerIdentifier as string}
+                    type="String"
+                    variableName="spec.provisionerIdentifier"
+                    showRequiredField={false}
+                    showDefaultField={false}
+                    allowedValuesType={ALLOWED_VALUES_TYPE.TEXT}
+                    showAdvanced={true}
+                    onChange={
+                      /* istanbul ignore next */ value => {
+                        setFieldValue('spec.provisionerIdentifier', value)
+                      }
+                    }
+                    isReadonly={readonly}
+                  />
+                )}
+              </div>
+            </>
+          )
+        }}
+      </Formik>
+    </>
+  )
+}
+
+const TerraformCloudRollbackInputStep: React.FC<TerraformCloudRollbackProps> = ({
+  template,
+  readonly,
+  path,
+  allowableTypes,
+  stepViewType
+}) => {
+  const { getString } = useStrings()
+  const { expressions } = useVariablesExpression()
+  const prefix = isEmpty(path) ? '' : `${path}.`
+  return (
+    <>
+      {isValueRuntimeInput(template?.timeout) && (
+        <div className={cx(stepCss.formGroup, stepCss.md)}>
+          <FormMultiTypeDurationField
```

Comment:
```suggestion
          < TimeoutFieldInputSetView
```
To support allowed values for timeout field.

Reply:
sure, will update in next PR of terraform cloud run step